src/cli/split-docs/index.ts

```diff
@@ -0,0 +1,377 @@
+  obj: any,
+  defName: string,
+  filename: string,
+  schemaFiles: any
+) {
+  if (!obj.discriminator) return;
+
+  const defPtr = `#/${COMPONENTS}/${OPENAPI3_COMPONENT.Schemas}/${defName}`;
+  const implicitMapping = {} as any;
+  for (const [name, { inherits, filename: parentFilename }] of Object.entries(schemaFiles) as any) {
+    if (inherits.indexOf(defPtr) > -1) {
+      const res = path.relative(path.dirname(filename), parentFilename);
+      implicitMapping[name] = res.startsWith('.') ? res : '.' + path.sep + res;
+    }
+  }
+
+  if (isNotObjectKeys(implicitMapping)) return;
+  const discriminatorPropSchema = obj.properties[obj.discriminator.propertyName];
+  const discriminatorEnum = discriminatorPropSchema && discriminatorPropSchema.enum;
+
+  const mapping = (obj.discriminator.mapping = obj.discriminator.mapping || {});
+  for (const name of Object.keys(implicitMapping)) {
+    if (discriminatorEnum && !discriminatorEnum.includes(name)) { continue; }
+    if (mapping[name] && mapping[name] !== implicitMapping[name]) {
+      process.stderr.write(yellow(
+        `warning: explicit mapping overlaps with local mapping entry ${red(name)} at ${blue(filename)}, Check manually, please`
+      ));
+    }
+    mapping[name] = implicitMapping[name];
+  }
+}
+
+function isNotSecurityComponentType(componentType: string) {
+  return componentType !== OPENAPI3_COMPONENT.SecuritySchemes
+}
+
+function findComponentTypes(components: any) {
+  return OPENAPI3_COMPONENTS
+    .filter(item => isNotSecurityComponentType(item) && Object.keys(components).includes(item))
+    .map(type => ({ name: type, data: components[type] }))
+}
+
+function isFileNotEqual(filename: string, componentData: string, componentName: string) {
+  const result = fs.existsSync(filename) && !isEqual(readYaml(filename), componentData);
+  if (result) {
+    process.stderr.write(yellow(
+      `warning: conflict for ${componentName} - file already exists
+         with different content: ${blue(filename)} ... Skip.\n`
+    ));
+  }
+  return result;
+}
+
+function cleanupComponentsExceptSecuritySchemes(
+  openapi: Oas3Definition,
+  componentType: string,
+  componentName: string
+) {
+  if (isNotSecurityComponentType(componentType)) {
+    // security schemas must referenced from components
+    // @ts-ignore
+    delete openapi.components[componentType][componentName];
+  }
+}
+
+function cleanupEmptyProperties(openapi: Oas3Definition, componentType: string) {
+  // @ts-ignore
+  if (isNotObjectKeys(openapi.components && openapi.components[componentType])) {
+    // @ts-ignore
+    delete openapi.components[componentType];
+  }
+  if (isNotObjectKeys(openapi.components)) {
+    delete openapi.components;
+  }
+}
+
+function createComponentDir(componetDirPath: string, componentType: string) {
+  if (isNotSecurityComponentType(componentType)) {
+    mkdirp.sync(componetDirPath);
+  }
+}
+
+function isRefComponentsExist(componentRefs: string[], components: string[]) {
```

Comment:
This name is awkward. Is___exist doesn’t work in English. Some valid variants:

Has_____
Is_______

(For example “hasRefComponents”)